### DISCUSSION TITLE
Noobie questions

### DISCUSSION THREAD

**Discussion by philipdanielhayton** [patrickdemooij9/SeoToolkit.Umbraco#90] [Q&A]
Hi Patrick,

Firstly thanks for this package, it looks super userful. I'm just playing around with it for the first time so apologies if I'm asking silly questions. but I was wondering if you could help with the following:

- Is there anyway to customize the format of the Meta Title, for example to add a dash and a sitename: "Some Page - My Website"
- Is there anyway to automatically crop the OG and Twitter card images down to the correct dimensions?
- Is there anywhere I can see all the options that can be set via appsettings?

Thanks again for this awesome package 👍🏻 and any help is greatly appreciated.

**Comment by patrickdemooij9** [comment]
Hi @philipdanielhayton,
First off, there are no noobie questions! It just means that either my documentation for it is missing or that the feature isn't there yet.

**Is there anyway to customize the format of the Meta Title, for example to add a dash and a sitename: "Some Page - My Website"**
Sort of. There is a way to do this through code by using the event 'BeforeMetaTags' (https://github.com/patrickdemooij9/SeoToolkit.Umbraco/blob/main/src/SeoToolkit.Umbraco.MetaFields.Core/Providers/DefaultMetaTagsProvider.cs#L27) and providing it with your own value, but that isn't ideal. I've created a new issue here: https://github.com/patrickdemooij9/SeoToolkit.Umbraco/issues/88 so that I can pick this up in the next release. It was something that was already on my mind to pick up.

**Is there anyway to automatically crop the OG and Twitter card images down to the correct dimensions?**
Currently not, but that is a good improvement to the current functionality. Created an issue for that as well: https://github.com/patrickdemooij9/SeoToolkit.Umbraco/issues/89

**Is there anywhere I can see all the options that can be set via appsettings?**
There wasn't before, but I have updated the documentation with all the information about appsettings: https://app.gitbook.com/s/fcrAqnWRQs0h6U4YJ9yD/settings/appsettings. If you are missing any, then please let me know.

And thank you for the feedback. If you have any more questions/feedback, then I would love to hear them.

I'll also move this to a discussion as I think it is a bit more fitting there than being an issue.

**Comment by philipdanielhayton** [comment]
Thanks so much for the swift (and brilliant) response Patrick 🙏🏻

**Comment by philipdanielhayton** [comment]
Hi Patrick, sorry it's me again, I'm just wondering what the best way to hook into the BeforeMetaTags event is? Does it raise an Umbraco notification or do I need to register an event listener via some other means?

Again your help is greatly appreciated 🙏🏻

**Reply by patrickdemooij9** [reply]
Hi @philipdanielhayton 

You should be able to inject an interface called IMetaTagsProvider and use the event on that one.

However, now that I am looking at the code, I am not 100% sure this will work as the lifecycle is set to Transient. So the instance sending the event is not the same as the one receiving it.

Unfortunately I am currently away on vacation. So if above doesn't work, I'll make a fix when I get back. I'll probably move it to a notification then to be more in line with the way Umbraco does it

**Comment by philipdanielhayton** [comment]
Ah buen vacaciones!

Ok no problem, we can make do for the time being. Make sure to send me a link when it's fixed so I can buy you a coffee! 🍻

**Reply by patrickdemooij9** [reply]
Hi @philipdanielhayton,
I've returned! I changed the event to a notification in the latest update (2.1.0, released today). You can find some more documentation about it here: https://seotoolkit.gitbook.io/useotoolkit/extensions/notifications#aftermetatagsnotification

If you really want to buy me a coffee, I have a buymeacoffee link: https://www.buymeacoffee.com/HHIl3uvWKj. But only if you really want, don't feel pressured!

Also, I would love to hear about your experience with SeoToolkit so far. Do you find it easy to use? Are there any features that think are missing?

**Reply by philipdanielhayton** [reply]
Amazing thanks Patrick, I'll update and test it later.

And yes I absolutely love Seo Toolkit, I've been able to remove a lot of custom code + third party packages. The only improvement I can think of would be for the OG & Twitter images to be auto cropped to the correct size, but other than that it's amazing my man. Sterling work 🍻
